docs/eventsV2.md

```diff
@@ -0,0 +1,268 @@
+ handler_name | text                        | not null
+ event_space  | text                        | not null
+ created_at   | timestamp without time zone | not null default now()
+```
+
+`scheduling_rules_type` values can be `pause` or `block`.
+
+## How features are implemented
+
+### Basic operation
+
+`emit` saved the event in the DB, and sends a notification to PubSub. QueueWorkers
+fetch notifications from PubSub, load the event, execute it, then delete the Event
+from PubSub and the DB.
+
+### Pausing/Unpausing
+
+**FSTODO**: Not implemented yet
+When the notification is delivered from PubSub to a QueueWorker, the QueueWorker
+checks the scheduling rules for the handler. If it is blocked by an admin or paused
+by the user, the PubSub notification is dropped, but the event remains in the DB.
+When the handler is unpaused, notifications are added to PubSub to schedule them.
+
+If the user pauses and unpauses in quick succession, there could be multiple
+notifications in PubSub for the same event. As a result, there is per-event locking
+using the `lockedAt` column.
+
+### Errors
+
+If a handler completes, then the Event is completed regardless of whether the outcome
+is an error.
+
+If there is an execution error, the QueueWorker will immediately tell PubSub to retry
+in 5 minutes.
+
+If there was an operational error, the QueueWorker will not be able to do anything to
+the event. However, the QueueWorker will not acknowledge the PubSub notification, and
+so when the PubSub ack deadline for that notification passes, PubSub will attempt to
+deliver it again. PubSub is configured to try a message 5 times, then it publishes it
+in a deadletter queue. We do not currently have any automation on the deadletter
+queue.
+
+An event that has more than 5 delays will be deleted by the QueueWorker.
+
+"Bad" notifications (perhaps they can't be read by the queueworker) will be discarded
+by PubSub after a week.
+
+### Queue counts
+
+**FSTODO**: add an index
+We do a DB query for the number of events for that canvas/handler.
+
+### Emit
+
+Done in `LibEvent` via `emit`, or automatically via `CronChecker`. Calls
+`EventQueueV2.enqueue`. This adds a new value to the events table with:
+
+- `locked_at = NULL`
+- `enqueued_at = CURRENT_TIMESTAMP`
+
+Note that `CronChecker` does not use `events_v2` table information for Cron
+scheduling (eg determining if it has been an hour since the last Cron event). It does
+that separately in the `cron_records` table and just emits events on the appropriate
+schedule, leaving it to the queues to handler after that.
+
+Emit also adds a notification to the PubSub topic. This will be delivered to a
+QueueWorker to tell it to try fetching and running an event. After 5 retries, PubSub
+will drop the notification, sending it to a dead-letter queue (which at the moment
+has no UI or anything).
+
+### QueueWorker Execution
+
+`EventQueueV2.dequeue` fetches a notification from PubSub and runs the process to
```

Comment:
A quick sentence before this paragraph explaining that Notifications are a PubSub concept created when [the subscribers should handle the publication?] would be useful here